syntax = "proto3";

package user_routes;

service UserRoutes{
    rpc tryLogin(Credentials) returns (LoginConfirm);
    rpc checkProfile(ProfileRequest) returns (UserInfo);
    rpc sendFriendRequest(RequestInfo) returns (Status);
    rpc saveGameData(MapData) returns (Status);
    rpc undoGameData(MapData) returns (Status);
}

message Credentials{
    string username = 1;
    string password = 2;
    bool newAccount = 3;
}

message LoginConfirm{
    int32 status = 1;
    string token = 2;
}

message ProfileRequest{
    int32 srcID = 1;
    int32 userID = 2;
}

message UserInfo{
    int32 status = 1;
    string username = 2;
}

message RequestInfo{
    int32 srcID = 1;
    int32 destID = 2;
}

message Status{
    int32 status = 1;
}

message MapData{
    int32 status = 1;
    repeated PlayerData nations = 2;
}

message PlayerData{
    int32 population = 1;
    repeated int32 provinceIDs = 2;
}